app/containers/SongContainer/index.js

```diff
@@ -0,0 +1,206 @@
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import get from 'lodash/get';
+import debounce from 'lodash/debounce';
+import isEmpty from 'lodash/isEmpty';
+import { Card, Skeleton, Input } from 'antd';
+import styled from 'styled-components';
+import { injectIntl } from 'react-intl';
+import { injectSaga } from 'redux-injectors';
+
+import React, { useEffect, useState } from 'react';
+
+import { selectSongContainer, selectSongsData, selectSongsError, selectQuery } from './selectors';
+import { songContainerCreators } from './reducer';
+import songContainerSaga from './saga';
+import T from '@components/T';
+import AlbumArt from '@app/components/AlbumArt';
+
+const { Search } = Input;
+const { Meta } = Card;
+
+const CustomCard = styled(Card)`
+  && {
+    width: ;
+    display: flex;
+    flex-wrap: wrap;
+    color: ${(props) => props.color};
+  }
+`;
+
+const SongCard = styled(Card)`
+  && {
+    margin: 1em;
+  }
+`;
+
+const Container = styled.div`
+  && {
+    display: flex;
+    flex-wrap: wrap;
+    justify-content: space-evenly;
+    max-width: ${(props) => props.maxwidth}px;
+    padding: ${(props) => props.padding}px;
+  }
+`;
+const ResultContainer = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    max-width: 80vw;
+    width: 100%;
+    margin: 20px auto;
+    padding: ${(props) => props.padding}px;
+  }
+`;
+
+export function SongContainer({
+  dispatchSongs,
+  dispatchClearSongs,
+  intl,
+  songsData = {},
+  songsError = null,
+  query,
+  maxwidth,
+  padding
+}) {
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const loaded = get(songsData, 'results', null) || songsError;
+    if (loading && loaded) {
+      setLoading(false);
+    }
+  }, [songsData]);
+
+  useEffect(() => {
+    if (query != '' && songsData?.songs?.length) {
+      setLoading(true);
```

Comment:
we already have loading in the store, no need for adding another state variable for it. Just use the one in the redux store

Reply:
okay , will remove state loading